### PR TITLE
[zh] sync command-line-tools reference of kube-apiserver

### PR DESCRIPTION
**Pull request by howieyuen**
ref: https://github.com/kubernetes/website/issues/25247

```
36	33	content/en/docs/reference/command-line-tools-reference/kube-apiserver.md
```

### CODE REVIEW

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 1]
```diff
@@ -1,227 +1,2642 @@
...
+--max-mutating-requests-inflight int&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;默认值：200
+</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+The maximum number of mutating requests in flight at a given time. 
+When the server exceeds this, it rejects requests. Zero for no limit.
+-->
+在给定时间内进行中可变请求的最大数量。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:1797]:
```suggestion
在给定时间内进行中变更性质请求的最大个数。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 2]
```diff
@@ -1,227 +1,2642 @@
...
+--max-requests-inflight int&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;默认值：400
+</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+The maximum number of non-mutating requests in flight at a given time. 
+When the server exceeds this, it rejects requests. Zero for no limit.
+-->
+在给定时间内进行中不可变请求的最大数量。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:1817]:
```suggestion
在给定时间内进行中非变更类型请求的最大数量。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 3]
```diff
@@ -1,227 +1,2642 @@
...
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--  
+An optional field indicating the minimum number of seconds a handler must 
+keep a request open before timing it out. Currently only honored by the 
+watch request handler, which picks a randomized value above this number 
+as the connection timeout, to spread out load.
+-->
+一个可选字段，表示一个 handler 在一个请求超时前，必须保持它处于打开状态的最小秒数。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:1839]:
```suggestion
可选字段，表示处理程序在请求超时前，必须保持其处于打开状态的最小秒数。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 4]
```diff
@@ -1,227 +1,2642 @@
...
+当前只对监听请求 handler 有效，它基于这个值选择一个随机数作为连接超时值，以达到分散负载的目的。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:1840]:
```suggestion
当前只对监听（Watch）请求的处理程序有效，它基于这个值选择一个随机数作为连接超时值，以达到分散负载的目的。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 5]
```diff
@@ -1,227 +1,2642 @@
...
+<tr>
+<td colspan="2">--oidc-ca-file string</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+If set, the OpenID server's certificate will be verified by one of 
+the authorities in the oidc-ca-file, otherwise the host's root CA set will be used.
+-->
+如果设置该值，将会使用 oidc-ca-file 中的任意一个 authority 对 OpenID 服务的证书进行验证，
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:1853]:
```suggestion
如果设置该值，将会使用 oidc-ca-file 中的机构之一对 OpenID 服务的证书进行验证，
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 6]
```diff
@@ -1,227 +1,2642 @@
...
+
+<tr>
+<td colspan="2">--oidc-client-id string</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+The client ID for the OpenID Connect client, must be set if oidc-issuer-url is set.
+-->
+使用 OpenID 连接的客户端的 ID，如果设置了 oidc-issuer-url，则必须设置这个值。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:1866]:
```suggestion
OpenID 连接客户端的要使用的客户 ID，如果设置了 oidc-issuer-url，则必须设置这个值。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 7]
```diff
@@ -1,227 +1,2642 @@
...
+identifier in "iss" claim of issued tokens. This value is a string or URI. 
+If this option is not a valid URI per the OpenID Discovery 1.0 spec, 
+the ServiceAccountIssuerDiscovery feature will remain disabled, even if 
+the feature gate is set to true. It is highly recommended that this value 
+comply with the OpenID spec: https://openid.net/specs/openid-connect-discovery-1_0.html. 
+In practice, this means that service-account-issuer must be an https URL. 
+It is also highly recommended that this URL be capable of serving OpenID 
+discovery documents at {service-account-issuer}/.well-known/openid-configuration.
+-->
+服务帐户令牌发行者的标识符。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2225]:
```suggestion
服务帐号令牌颁发者的标识符。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 8]
```diff
@@ -1,227 +1,2642 @@
...
+发行者将在已发行令牌的 "iss" 声明中声明此标识符。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2226]:
```suggestion
颁发者将在已办法令牌的 "iss" 声明中检查此标识符。 
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 9]
```diff
@@ -1,227 +1,2642 @@
...
+the feature gate is set to true. It is highly recommended that this value 
+comply with the OpenID spec: https://openid.net/specs/openid-connect-discovery-1_0.html. 
+In practice, this means that service-account-issuer must be an https URL. 
+It is also highly recommended that this URL be capable of serving OpenID 
+discovery documents at {service-account-issuer}/.well-known/openid-configuration.
+-->
+服务帐户令牌发行者的标识符。
+发行者将在已发行令牌的 "iss" 声明中声明此标识符。 
+此值为字符串或 URI。
+如果根据 OpenID Discovery 1.0 规范此选项不是有效的 URI，则即使功能门控设置为 true，
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2228]:
```suggestion
如果根据 OpenID Discovery 1.0 规范检查此选项不是有效的 URI，则即使特性门控设置为 true，
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 10]
```diff
@@ -1,227 +1,2642 @@
...
+ServiceAccountIssuerDiscovery 功能也将保持禁用状态。 
+强烈建议该值符合 OpenID 规范：https://openid.net/specs/openid-connect-discovery-1_0.html。 
+实际上，这意味着服务帐户发行者必须是 https URL。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2231]:
```suggestion
实践中，这意味着 service-account-issuer 取值必须是 HTTPS URL。 
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 11]
```diff
@@ -1,227 +1,2642 @@
...
+discovery documents at {service-account-issuer}/.well-known/openid-configuration.
+-->
+服务帐户令牌发行者的标识符。
+发行者将在已发行令牌的 "iss" 声明中声明此标识符。 
+此值为字符串或 URI。
+如果根据 OpenID Discovery 1.0 规范此选项不是有效的 URI，则即使功能门控设置为 true，
+ServiceAccountIssuerDiscovery 功能也将保持禁用状态。 
+强烈建议该值符合 OpenID 规范：https://openid.net/specs/openid-connect-discovery-1_0.html。 
+实际上，这意味着服务帐户发行者必须是 https URL。 
+还强烈建议此 URL 能够在 {service-account-issuer}/.well-known/openid-configuration 处提供 OpenID 发现文档。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2232]:
```suggestion
还强烈建议此 URL 能够在 {service-account-issuer}/.well-known/openid-configuration
处提供 OpenID 发现文档。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 12]
```diff
@@ -1,227 +1,2642 @@
...
+<!--
+Overrides the URI for the JSON Web Key Set in the discovery doc served at 
+/.well-known/openid-configuration. This flag is useful if the discovery 
+docand key set are served to relying parties from a URL other than the 
+API server's external (as auto-detected or overridden with external-hostname). 
+Only valid if the ServiceAccountIssuerDiscovery feature gate is enabled.
+-->
+覆盖 /.well-known/openid-configuration 提供的发现文档中 JSON Web 密钥集的 URI。
+如果发现文档和密钥集是通过 API 服务器外部
+（而非自动检测到或被外部主机名覆盖）以外的URL提供给依赖方的，则此标志很有用。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2250]:
```suggestion
（而非自动检测到或被外部主机名覆盖）之外的 URL 提供给依赖方的，则此标志很有用。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 13]
```diff
@@ -1,227 +1,2642 @@
...
+仅在启用 ServiceAccountIssuerDiscovery 功能门控的情况下有效。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2251]:
```suggestion
仅在启用 ServiceAccountIssuerDiscovery 特性门控的情况下有效。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 14]
```diff
@@ -1,227 +1,2642 @@
...
+File containing PEM-encoded x509 RSA or ECDSA private or public keys,
+used to verify ServiceAccount tokens. The specified file can contain 
+multiple keys, and the flag can be specified multiple times with 
+different files. If unspecified, --tls-private-key-file is used. 
+Must be specified when --service-account-signing-key is provided
+-->
+包含 PEM 编码的 x509 RSA 或 ECDSA 私钥或公钥的文件，用于验证 ServiceAccount 令牌。
+指定的文件可以包含多个键，并且可以使用不同的文件多次指定标志。
+如果未指定，则使用 --tls-private-key-file。
+提供 --service-account-signing-key 时必须指定
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2270]:
```suggestion
提供 --service-account-signing-key 时必须指定。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 15]
```diff
@@ -1,227 +1,2642 @@
...
+-->
+--service-account-lookup&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;默认值：true
+</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+If true, validate ServiceAccount tokens exist in etcd as part of authentication.
+-->
+如果为 true，则在身份验证中验证 etcd 中是否存在 ServiceAccount 令牌。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2287]:
```suggestion
如果为 true，则在身份认证时验证 etcd 中是否存在 ServiceAccount 令牌。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 16]
```diff
@@ -1,227 +1,2642 @@
...
+<td colspan="2">--service-account-max-token-expiration duration</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+The maximum validity duration of a token created by the service account token issuer. 
+If an otherwise valid TokenRequest with a validity duration larger than this value is requested, 
+a token will be issued with a validity duration of this value.
+-->
+服务帐户令牌发行者创建的令牌的最大有效期。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2301]:
```suggestion
服务帐户令牌发布者创建的令牌的最长有效期。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 17]
```diff
@@ -1,227 +1,2642 @@
...
+如果请求有效期大于此值的有效令牌请求，将使用此值的有效期发行令牌。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2302]:
```suggestion
如果请求有效期大于此值的有效令牌请求，将使用此值的有效期颁发令牌。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 18]
```diff
@@ -1,227 +1,2642 @@
...
+<td colspan="2">--service-account-signing-key-file string</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+Path to the file that contains the current private key of the service account token issuer. 
+The issuer will sign issued ID tokens with this private key. 
+(Requires the 'TokenRequest' feature gate.)
+-->
+包含服务帐户令牌发行者当前私钥的文件的路径。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2316]:
```suggestion
包含服务帐户令牌颁发者当前私钥的文件的路径。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 19]
```diff
@@ -1,227 +1,2642 @@
...
+发行者将使用此私钥签署已发行的 ID 令牌。（需要开启 "TokenRequest" 功能门控。）
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2317]:
```suggestion
颁发者将使用此私钥签署所颁发的 ID 令牌（需要启用 "TokenRequest" 特性门控）。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 20]
```diff
@@ -1,227 +1,2642 @@
...
+<tr>
+<td colspan="2">--service-cluster-ip-range string</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+A CIDR notation IP range from which to assign service cluster IPs. 
+This must not overlap with any IP ranges assigned to nodes or pods.
+-->
+CIDR 表示的 IP 范围，服务的 cluster ip 将从中分配。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2330]:
```suggestion
CIDR 表示的 IP 范围用来为服务分配集群 IP。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 21]
```diff
@@ -1,227 +1,2642 @@
...
+此地址不得与分配给节点或 Pod 的任何 IP 范围重叠。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2331]:
```suggestion
此地址不得与指定给节点或 Pod 的任何 IP 范围重叠。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 22]
```diff
@@ -1,227 +1,2642 @@
...
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+The previous version for which you want to show hidden metrics. Only the 
+previous minor version is meaningful, other values will not be allowed. 
+The format is &lt;major&gt;.&lt;minor&gt;, e.g.: '1.16'. The purpose of 
+this format is make sure you have the opportunity to notice if the next
+release hides additional metrics, rather than being surprised when they 
+are permanently removed in the release after that.
+-->
+你要显示隐藏指标的先前版本。仅先前的次要版本有意义，将不允许其他值。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2367]:
```suggestion
你要显示隐藏指标的先前版本。仅先前的次要版本有意义，不允许其他值。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 23]
```diff
@@ -1,227 +1,2642 @@
...
+-->
+--stderrthreshold severity&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;默认值：2
+</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;">
+<!--
+logs at or above this threshold go to stderr
+-->
+达到或超过此阈值的日志转到 stderr
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2429]:
```suggestion
将达到或超过此阈值的日志写到标准错误输出
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 24]
```diff
@@ -1,227 +1,2642 @@
...
+domain names, possibly with prefixed wildcard segments. The domain 
+patterns also allow IP addresses, but IPs should only be used if 
+the apiserver has visibility to the IP address requested by a client. 
+If no domain patterns are provided, the names of the certificate are 
+extracted. Non-wildcard matches trump over wildcard matches, explicit
+domain patterns trump over extracted names. For multiple key/certificate 
+pairs, use the --tls-sni-cert-key multiple times. Examples: 
+"example.crt,example.key" or "foo.crt,foo.key:*.foo.com,foo.com".
+-->
+一对 x509 证书和私钥文件路径，（可选）后缀为完全合格域名的域名模式列表，可能带有前缀的通配符段。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2543]:
```suggestion
一对 x509 证书和私钥文件路径，（可选）后缀为全限定域名的域名模式列表，
可以使用带有通配符的前缀。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 25]
```diff
@@ -1,227 +1,2642 @@
...
+域模式也允许使用 IP 地址，但仅当 apiserver 对客户端请求的IP地址具有可见性时，才应使用 IP。
+如果未提供域模式，则提取证书的名称。
+非通配符匹配胜过通配符匹配，显式域模式胜过提取名称。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2546]:
```suggestion
非通配符匹配优先于通配符匹配，显式域模式优先于提取出的名称。
```

**Comment by tengqm** [comment]
@howieyuen Due to github limitations, cannot finish the review of the first doc. Please try fix the issues and commit again. I have to continue the review on a new commit because github cannot open a page with too many review comments.

**Comment by tengqm** [comment]
Another suggestion is to split this PR so that each is of a reasonable size.

**Comment by howieyuen** [comment]
> Another suggestion is to split this PR so that each is of a reasonable size.

ok, I after commit these changes, I will split into 2 PRs

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 26]
```diff
@@ -1,227 +1,2649 @@
...
+<!--
+Watch cache size settings for some resources (pods, nodes, etc.), comma separated. 
+The individual setting format: resource[.group]#size, where resource is lowercase 
+plural (no version), group is omitted for resources of apiVersion v1 (the legacy 
+core API) and included for others, and size is a number. It takes effect when 
+watch-cache is enabled. Some resources (replicationcontrollers, endpoints, nodes,
+pods, services, apiservices.apiregistration.k8s.io) have system defaults set by 
+heuristics, others default to default-watch-cache-size
+-->
+观察某些资源（容器，节点等）的缓存大小设置，以逗号分隔。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2638]:
```suggestion
某些资源（pods、nodes 等）的监视缓存大小设置，以逗号分隔。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 27]
```diff
@@ -1,227 +1,2649 @@
...
+单个设置格式：resource[.group]#size，其中资源为小写复数（无版本），
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2639]:
```suggestion
每个资源对应的设置格式：resource[.group]#size，其中 resource 为小写复数（无版本），
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 28]
```diff
@@ -1,227 +1,2649 @@
...
+apiVersion v1（旧版核心 API）的资源省略了 group，
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2640]:
```suggestion
对于 apiVersion v1（旧版核心 API）的资源要省略 group，
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 29]
```diff
@@ -1,227 +1,2649 @@
...
+其他资源包括了 group，大小为一个数字。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2641]:
```suggestion
对其它资源要给出 group，size 为一个数字。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 30]
```diff
@@ -1,227 +1,2649 @@
...
+启用监视高速缓存时，此功能生效。
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2642]:
```suggestion
启用 watch-cache 时，此功能生效。
```

File: content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 31]
```diff
@@ -1,227 +1,2649 @@
...
+某些资源（replicationcontrollers, endpoints, nodes, pods, services, apiservices.apiregistration.k8s.io）
```
Comment (tengqm) [inline content/zh/docs/reference/command-line-tools-reference/kube-apiserver.md:2643]:
```suggestion
某些资源（replicationcontrollers、endpoints、nodes、pods、services、apiservices.apiregistration.k8s.io）
```

**Pull request merged**